platform/packages/platform/src/contract.rs

```diff
@@ -1,24 +1,59 @@
-use sdk::cosmwasm_std::{Addr, ContractInfoResponse, QuerierWrapper, WasmQuery};
+use sdk::{
+    cosmwasm_std::{Addr, CodeInfoResponse, ContractInfoResponse, QuerierWrapper, WasmQuery},
+    schemars::{self, JsonSchema},
+};
+use serde::{Deserialize, Serialize};
 
 use crate::{error::Error, result::Result};
 
 pub type CodeId = u64;
 
+#[derive(Serialize, Deserialize, Clone, Copy, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+#[serde(transparent)]
```

Comment:
We might also like to put `#[repr(transparent)]` to tell the compiler to inherit the inner type's memory layout. In the best case it would provide us with an improvement and in the worst case nothing would change.

Reply:
if there is no reason,would prefer not to tell the compiler how to represent it better in the memory

Reply:
There is a reason, and it's optimization. If `NonZero*` type is plugged inside it can help when it's used inside an `Option`. Instead of being two "words" it would be just one. It's something we'd generally want on single field/single non-zero-type field structs.

Reply:
I follow the best practices on optimizations and do not do premature ones.
Just to note here that we are talking about `#repr` and not about NonZero*

Reply:
They are closely related if you think about it. Even if we want to call it a premature optimization, it is a zero cost one. The worst case is that we'd add a new non-zero-sized field, which is highly unlikely for such specialized type, that would require removing it.